pkg/controller/mongodb/mongodb_controller.go

```diff
@@ -251,15 +277,17 @@ func (r *ReplicaSetReconciler) createOrUpdateStatefulSet(mdb mdbv1.MongoDB) erro
 	return nil
 }
 
-// setAnnotation updates the monogdb resource with the given namespaced name and sets the annotation
+// setAnnotations updates the monogdb resource with the given namespaced name and sets the annotation
 // "key" with the provided value "val"
-func (r ReplicaSetReconciler) setAnnotation(nsName types.NamespacedName, key, val string) error {
+func (r ReplicaSetReconciler) setAnnotations(nsName types.NamespacedName, annotations map[string]string) error {
```

Comment:
As we add more annotations I feel like this function is going to get a bit hard to use. To avoid overwriting existing annotations we'll probably have to always update `mdb.Annotations` directly and then pass it to the function. If we do this we'll also need a way to deal with the case when `Annotations` is nil.

One idea is to introduce a new function on the MongoDB type:
```
func (m *MongoDB) SetAnnotation(key, value, string) {
  if m.Annotations == nil {
    m.Annotations = map[string]string{}
  }

  m.Annotation[key] = value
}
```

And then change `setAnnotations` to `ReplicaSetReconciler.saveAnnotations(mdb mdbv1.MongoDB)` to persist the changes. This way we don't have to worry about overwriting. Thoughts?

Reply:
I think this is a good point, maybe we can do a smaller follow up PR to avoid clutter in this one.

Reply:
Good :)